src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImageFunctionalTest.groovy

```diff
@@ -95,6 +95,16 @@ class DockerBuildImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
         noExceptionThrown()
     }
 
+    def "can build image with different targets"() {
+        buildFile << buildMultiStageImage()
+
+        when:
+        BuildResult result = build('buildTarget')
+
+        then:
+        result.output.contains("Successfully built")
```

Comment:
I was thinking more in terms of actually inspecting the built image. There's no point in verifying the console output here to see if the build was successful as the `build` method would throw an exception if the build failed. You will be able to find similar example if you look through some of the other functional tests.

Reply:
additional checks added